ext/pcre/php_pcre.c

```diff
@@ -1135,6 +1135,11 @@ static void php_do_pcre_match(INTERNAL_FUNCTION_PARAMETERS, int global) /* {{{ *
 		RETURN_FALSE;
 	}
 
+	if (start_offset == ZEND_LONG_MIN) {
+		zend_argument_value_error(5, "must be greater than " ZEND_LONG_FMT, ZEND_LONG_MIN);
+		RETURN_THROWS();
+	}
```

Comment:
Surely it should also be checked that it is not larger than ZEND_LONG_MAX ?

Reply:
How long can be larger than ZEND_LONG_MAX?

Reply:
@mvorisek is right on this, start_offset is a zend_long.